codegenerator/cli/src/fuel/abi.rs

```diff
@@ -98,7 +101,7 @@ impl FuelAbi {
             .filter_map(|type_decl| {
                 let generic_param_name =
                     extract_value_after_keyword("generic", &type_decl.type_field)?;
-                Some((type_decl.type_id, generic_param_name))
+                Some((type_decl.type_id, generic_param_name.uncapitalize()))
```

Comment:
Is it needed for any cases or just trying to be idiomatic with rescript?

2 Reasons I think we shouldn't uncapitalized
1. `a` and `A` would be valid param names treated as separate parameters (we would be collapsing this case)
2. Typescript generics are typically capitalised so we would be trading off idiomatic TS for idiomatic rescript

Reply:
When we try to get a schema for the generic like `<genericName>Schema` it fails because of invalid var name. Maybe we can prefix it for the case to be `generic<genericName>Schema`

Reply:
This logic is not bullet proof at all. I just made it compile for the greeter template I use for tests

Reply:
Ok cool, I'll leave it to you 👍🏼